### PR TITLE
feat: new `resolveId` options

### LINKED ISSUE #69: support resolving full paths
* toExports
* toImports

(For pnpm/pnp support.)

### PR DESCRIPTION
**Pull request by antfu**
close #69 

A custom function for integrations to resolve the auto import

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export interface UnimportContext {
   map: Map<string, Import>
   addons: Addon[]
   invalidate(): void
+  resolveId(id: string, parentId?: string): Thenable<string | void>
```
Comment (pi0) [inline src/types.ts:40]: Small note: Rollup i think uses `null` as fallback

File: src/types.ts [hunk 2]
```diff
@@ -74,11 +79,17 @@ export interface TypeDeclrationOptions {
...
   /**
    * @default false
    */
   mergeExisting?: boolean
+
+  /**
+   * Inject using `import` or require
+   * @default auto-detection
+   */
+  isCJS?: boolean
```
Comment (pi0) [inline src/types.ts:92]: Why we would need this feature?

**Comment by pi0** [comment]
This seems a very nice addition to support resolvedId with maximum flexibility. But maybe we can have built-in resolver with paths option support? At least it makes implementation between nuxt app and nitro server consistent in one place.

File: src/types.ts (hunk 2 above)
Comment (antfu) [reply src/types.ts]: Was trying to have the resolve logic inside the utils, removed.

File: src/utils.ts [hunk 3]
```diff
@@ -232,3 +232,9 @@ export function normalizeImports (imports: Import[]): Import[] {
   }
   return imports
 }
+
+export function resolveIdAbsolute (id: string, parentId?: string) {
```
Comment (antfu) [inline src/utils.ts:236]: Exposed a util here

**Pull request merged**